service/storage_proxy.cc

```diff
@@ -1451,7 +1474,9 @@ class abstract_write_response_handler : public seastar::enable_shared_from_this<
     future<> apply_remotely(gms::inet_address ep, inet_address_vector_replica_set&& forward,
             storage_proxy::response_id_type response_id, storage_proxy::clock_type::time_point timeout,
             tracing::trace_state_ptr tr_state) {
-        return _mutation_holder->apply_remotely(*_proxy, ep, std::move(forward), response_id, timeout, std::move(tr_state), _rate_limit_info);
+        return _mutation_holder->apply_remotely(*_proxy, ep, std::move(forward),
+            response_id, timeout, std::move(tr_state), _rate_limit_info,
+            {_effective_replication_map_ptr->get_token_metadata().get_version()});
```

Comment:
Shouldn't we also do fencing on local writes? If we do drain with timeout a local write may complete after the fence moved.